schema.json

```diff
@@ -1,255 +1,266 @@
+          "minimum": 0
+        },
+        "maxItems": {
+          "description": "The maximum list size.",
           "type": "integer",
-          "maximum": 4294967295,
           "minimum": 0
         }
       },
-      "required": ["label", "value"]
+      "required": [ "type", "itemType" ]
     },
-    "name": {
+    "nameType": {
+      "description": "A name is a string having a length of at least 1.",
       "type": "string",
       "minLength": 1
     },
-    "range": {
+    "oneOfTypes": {
+      "description": "One of any of the types. This provides a single location
+                      to keep the list.",
+      "oneOf": [
+        { "$ref": "#/definitions/bitFieldType" },
+        { "$ref": "#/definitions/compoundType" },
+        { "$ref": "#/definitions/integerType" },
+        { "$ref": "#/definitions/listType" },
+        { "$ref": "#/definitions/pdEnvelopeType" },
+        { "$ref": "#/definitions/simpleType" },
+        { "$ref": "#/definitions/stringType" },
+        { "$ref": "#/definitions/subdeviceType" }
+      ]
+    },
+    "subdeviceType": {
+      "description": "A subdevice type. One of the root device (e.g. value 0),
+                      the root or all subdevices or the broadcast subdevice,
+                      the root or any subdevice, or just any subdevice.",
+      "type" : "string",
+      "enum": [
+            "root",
+            "root_or_any_subdevice_or_broadcast",
+            "root_or_any_subdevice",
+            "any_subdevice"
+          ]
+    },
+    "pdEnvelopeType": {
+      "description": "Contains a length/data pair for one Parameter Data item,
+                      where the length is an unsigned 8-bit value and the data
+                      has 'length' bytes. This exists to provide a schema
+                      definition for the 'envelope' of a PDL/PD pair.",
+      "type": "object",
+      "properties": {
+        "type": { "const": "pdEnvelope" },
+        "length": {
+          "description": "The data length can be optionally specified.",
+          "type": "integer",
+          "minimum": 0
+        }
+      },
+      "required": [ "type" ]
+    },
+    "rangeType": {
+      "description": "Defines an inclusive range of numbers.",
       "type": "object",
       "properties": {
-        "lower": {
+        "minimum": {
+          "description": "The lower bound, inclusive.",
           "type": "integer"
         },
-        "upper": {
+        "maximum": {
+          "description": "The upper bound, inclusive.",
           "type": "integer"
         }
       },
-      "required": ["lower", "upper"]
+      "required": [ "minimum", "maximum" ]
+    },
+    "simpleType": {
+      "description": "Simple type that has no fields other than a name and
+                      type.",
+      "type": "object",
+      "properties": {
+        "name": { "$ref": "#/definitions/nameType" },
+        "type": {
+          "enum": [
+            "bool",
+            "hostname",
```

Comment:
I'm not sure if this should be in here, this list ought to match Table A-15 in E1.20, obviously hostname has additional restrictions so I don't know where it should be added or removed from.

Reply:
The intent is be able to specify random specific types, for example "true", "a.b.com", "192.168.1.1", with their intent. There's no way this can be exhaustive, sadly, but this list seems to accommodate all the needs of any RDM messages. This was here when I started working on the schema. Note that the values from Table A-15 are already accommodated in the other types, for example "integerType".

Reply:
Sorry, I mean this list should match the one in Table A-15 in E1.20, if they're valuable/relevant types they should be added to the master E1.20 standard too, you shouldn't only find out they exist by looking in the schema doc, otherwise taking your URI example it's not actually specified anywhere what that is or means.